src/values.rs

```diff
@@ -67,11 +76,68 @@ impl Type {
         use self::Type::*;
         match *self {
             Duration | TimeStamp | Temperature | ExtNumeric | Color => false,
-            Unit | Bool | String | Json | Binary => true,
+            Unit | String | Json | Binary | OnOff | OpenClosed | ExtBool => true,
+        }
+    }
+}
+
+/// An on/off state. Internal representation may be either On or Off.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub enum OnOff {
+    On,
+    Off,
+}
+
+impl OnOff {
+    fn as_bool(&self) -> bool {
+        match *self {
```

Comment:
I see two different approaches in the code ```match *self { ENTITY => .... }``` and ```match self { &ENTITY => ....}```. So I stick to the former because of [1]. There is no any explanation though :/

[1] https://github.com/rust-lang/rust/blob/master/src/doc/style/features/match.md